### PR TITLE
Status of quarantined tests is stored in Github Issue

### PR DESCRIPTION
**Pull request by potiuk**
This is an automated tracking solution for our Quarantined tests.

Thanks to it we will be able to track the "flakiness of our tests. The status of last 10 test executions for all quarantined runs are kept in in the issue https://github.com/apache/airflow/issues/10118 and once last 10 runs of a test are ""success" we can remove it from Quarantine. 

The quarantine test are run now in a separate workflow and repeated 4 times a day on schedule.

I hope this way we will be able to clean-up the tests quickly.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: scripts/ci/in_container/update_quarantined_test_status.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            test_id=test['classname'] + "::" + test['name'],
+            file=test['file'],
+            line=test['line'],
+            name=test['name'],
+            classname=test['classname'],
+            result=len(test.contents) == 0
+        ))
+
+    token = os.environ.get("GITHUB_TOKEN")
+    print(f"Token: {token}")
```
Comment (turbaszek) [inline scripts/ci/in_container/update_quarantined_test_status.py:202]: Should this be printed?

Comment (potiuk) [reply scripts/ci/in_container/update_quarantined_test_status.py:202]: Does not matter. It is masked anyway :)

Comment (potiuk) [reply scripts/ci/in_container/update_quarantined_test_status.py:202]: Just by seing ***** you know you get it right. girthub automatically mask github token when it is printed 😱

Comment (potiuk) [reply scripts/ci/in_container/update_quarantined_test_status.py:202]: See " https://github.com/apache/airflow/pull/10119/checks?check_run_id=938181654#step:6:1155

File: scripts/ci/in_container/update_quarantined_test_status.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    github_repository = os.environ.get('GITHUB_REPOSITORY')
+    if not github_repository:
+        raise Exception("Github Repository must be defined!")
+    user, repo = github_repository.split("/")
+    print(f"User: {user}, Repo: {repo}")
+    issue_id = int(os.environ.get('ISSUE_ID', 0))
```
Comment (ad-m) [inline scripts/ci/in_container/update_quarantined_test_status.py:208]: Have you considered using the title of the issue here? Some bots, eg renovate-bot, use issues as a title-based database (not create new pull request when previous one is closed without merge; create new pull request when expected not exist).

Comment (potiuk) [reply scripts/ci/in_container/update_quarantined_test_status.py:208]: There is no point in that. We have one single issue where we will keep status of all the tests. We are not going to open/close the issue - it's description is merely used as both - visualisation of status of quarantined issues as well as database where we keep status of last tests. So we can define one issue where we keep those issues. and we are only limiting that to apache/airflow repo (in all other repos the script is not executed.

File: scripts/ci/in_container/update_quarantined_test_status.py (hunk 1 above)
Comment (turbaszek) [reply scripts/ci/in_container/update_quarantined_test_status.py:202]: > girthub automatically mask github token when it is printed 😱

I didn't know that. If that's the case I see no problem

**Comment by turbaszek** [comment]
Should we add in #10118 some info about the process and the issue itself?

**Comment by potiuk** [comment]
Good Point. I will add it :)

**Review by turbaszek — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Review by feluelle — APPROVED** [review]
LGTM 👍 

Should we auto-close the issue if no quarantine tests left? And can we only run this action as long as the related issue is open? So it won't run extra actions.

**Comment by potiuk** [comment]
I've added a few things:

1) Documentation in CI.rst
2) Added header to the issue explaining how it gets updated.
3) added a "Comment" section that is preserved (so you can for example add link to issue if you are working on flaky test)
4) added configurable ISSUE_ID -> we need different set of tests for master/v1-10-test/v1-10-stable

@feluelle  -> I think we should not close it - flaky tests will happen sooner or later. But indeed closing/opening the issue might be a good idea indeed.

**Comment by potiuk** [comment]
I would still run the action anyway - it's not too costly one, and also we should re-open the issue if we add a new quarantined test - so the action (if at all) should be based on a number of tests that we mark as quarantined (so leaving it as is is best)

**Comment by potiuk** [comment]
@feluelle -> auto-closing/opening added  :)

**Pull request merged**